mod/dfrn_notify.php

```diff
@@ -53,148 +51,7 @@ function dfrn_notify_post(App $a) {
-	$importer = DFRN::getImporter($contact['id'], $user['uid']);
-
-	if ((($writable != (-1)) && ($writable != $importer['writable'])) || ($importer['forum'] != $forum) || ($importer['prv'] != $prv)) {
-		$fields = ['writable' => ($writable == (-1)) ? $importer['writable'] : $writable,
-			'forum' => $forum, 'prv' => $prv];
-		DBA::update('contact', $fields, ['id' => $importer['id']]);
-
-		if ($writable != (-1)) {
-			$importer['writable'] = $writable;
-		}
-		$importer['forum'] = $page;
-	}
-
-
-	// if contact's ssl policy changed, update our links
-
-	$importer = Contact::updateSslPolicy($importer, $ssl_policy);
-
-	Logger::log('data: ' . $data, Logger::DATA);
-
-	if ($dissolve == 1) {
-		// Relationship is dissolved permanently
-		Contact::remove($importer['id']);
-		Logger::log('relationship dissolved : ' . $importer['name'] . ' dissolved ' . $importer['username']);
-		System::xmlExit(0, 'relationship dissolved');
-	}
-
-	$rino = DI::config()->get('system', 'rino_encrypt');
-	$rino = intval($rino);
-
-	if (strlen($key)) {
-
-		// if local rino is lower than remote rino, abort: should not happen!
-		// but only for $remote_rino > 1, because old code did't send rino version
-		if ($rino_remote > 1 && $rino < $rino_remote) {
-			Logger::log("rino version '$rino_remote' is lower than supported '$rino'");
-			System::xmlExit(0, "rino version '$rino_remote' is lower than supported '$rino'");
-		}
-
-		$rawkey = hex2bin(trim($key));
-		Logger::log('rino: md5 raw key: ' . md5($rawkey), Logger::DATA);
-
-		$final_key = '';
-
-		if ($dfrn_version >= 2.1) {
-			if (($importer['duplex'] && strlen($importer['cprvkey'])) || !strlen($importer['cpubkey'])) {
-				openssl_private_decrypt($rawkey, $final_key, $importer['cprvkey']);
-			} else {
-				openssl_public_decrypt($rawkey, $final_key, $importer['cpubkey']);
-			}
-		} else {
-			if (($importer['duplex'] && strlen($importer['cpubkey'])) || !strlen($importer['cprvkey'])) {
-				openssl_public_decrypt($rawkey, $final_key, $importer['cpubkey']);
-			} else {
-				openssl_private_decrypt($rawkey, $final_key, $importer['cprvkey']);
-			}
-		}
-
-		switch ($rino_remote) {
-			case 0:
-			case 1:
-				// we got a key. old code send only the key, without RINO version.
-				// we assume RINO 1 if key and no RINO version
-				$data = DFRN::aesDecrypt(hex2bin($data), $final_key);
-				break;
-			default:
-				Logger::log("rino: invalid sent version '$rino_remote'");
-				System::xmlExit(0, "Invalid sent version '$rino_remote'");
-		}
-
-		Logger::log('rino: decrypted data: ' . $data, Logger::DATA);
-	}
-
-	Logger::log('Importing post from ' . $importer['addr'] . ' to ' . $importer['nickname'] . ' with the RINO ' . $rino_remote . ' encryption.', Logger::DEBUG);
-
-	$ret = DFRN::import($data, $importer, Conversation::PARCEL_LEGACY_DFRN, Conversation::PUSH);
-	System::xmlExit($ret, 'Processed');
-
-	// NOTREACHED
+	throw new HTTPException\BadRequestException();
```

Comment:
I believe we should add a deprecation message to this error.

Reply:
Who would read this? And AFAIK there is no "deprecated" HTTP error code, isn't it?

Reply:
I meant adding a meaningful message but I guess it isn't necessary.